File: PeptideHitResultsProcessor/clsSearchResultsMSPathFinder.vb
Option Strict On

' This class is used to track the peptide details for a MSPathFinder search result
' See clsSearchResultsBaseClass for additional information
'
' -------------------------------------------------------------------------------
' Written by Matthew Monroe for the Department of Energy (PNNL, Richland, WA)
' Created 7/16/2015
'
' E-mail: matthew.monroe@pnnl.gov or matt@alchemistmatt.com
' Website: http://omics.pnl.gov/ or http://www.sysbio.org/resources/staff/
' -------------------------------------------------------------------------------

Public Class clsSearchResultsMSPathFinder
    Inherits clsSearchResultsBaseClass

#Region "Classwide Variables"
    ' Note that "Automatic properties" are being used; thus, we don't need to explicitly define class variables 
#End Region

#Region "Properties"

    ' Auto-Properties
    Public Property MostAbundantIsotopeMz As String
    Public Property Composition As String
    Public Property ProteinDesc As String
    Public Property ProteinLength As String
    Public Property ResidueStart As String
    Public Property ResidueEnd As String
    Public Property MatchedFragments As String
    Public Property QValue As String
    Public Property PepQValue As String

#End Region

    Public Sub New(ByRef objPeptideMods As clsPeptideModificationContainer)
        ' Note that the following call will call both the base class's Clear sub and this class's Clear Sub
        MyBase.New(objPeptideMods)
    End Sub

    Public Overrides Sub Clear()
        MyBase.Clear()

        MostAbundantIsotopeMz = String.Empty
        Composition = String.Empty
        ProteinDesc = String.Empty
        ProteinLength = String.Empty
        ResidueStart = String.Empty
        ResidueEnd = String.Empty
        MatchedFragments = String.Empty
        QValue = String.Empty
        PepQValue = String.Empty

    End Sub



End Class

File: PeptideHitResultsProcessor/clsSearchResultsMSPathFinder.cs
// -------------------------------------------------------------------------------
// Written by Matthew Monroe for the Department of Energy (PNNL, Richland, WA)
//
// E-mail: matthew.monroe@pnnl.gov or proteomics@pnnl.gov
// Website: https://omics.pnl.gov/ or https://www.pnnl.gov/sysbio/ or https://panomics.pnnl.gov/
// -------------------------------------------------------------------------------
//
// Licensed under the 2-Clause BSD License; you may not use this file except
// in compliance with the License.  You may obtain a copy of the License at
// https://opensource.org/licenses/BSD-2-Clause
//
// Copyright 2018 Battelle Memorial Institute

using PHRPReader;

namespace PeptideHitResultsProcessor
{
    /// <summary>
    /// This class is used to track the peptide details for a MSPathFinder search result
    /// See clsSearchResultsBaseClass for additional information
    /// </summary>
    public class clsSearchResultsMSPathFinder : clsSearchResultsBaseClass
    {
        #region "Classwide Variables"
        // Note that "Automatic properties" are being used; thus, we don't need to explicitly define class variables
        #endregion

        #region "Properties"

        // Auto-Properties
        public string MostAbundantIsotopeMz { get; set; }
        public string Modifications { get; set; }
        public string Composition { get; set; }
        public string ProteinDesc { get; set; }
        public string ProteinLength { get; set; }
        public string ResidueStart { get; set; }
        public string ResidueEnd { get; set; }
        public string MatchedFragments { get; set; }
        public string SpecEValue { get; set; }
        public string EValue { get; set; }
        public string QValue { get; set; }
        public string PepQValue { get; set; }

        #endregion

        // Note that the following call will call both the base class's Clear method and this class's Clear method
        public clsSearchResultsMSPathFinder(clsPeptideModificationContainer peptideMods, clsPeptideMassCalculator peptideSeqMassCalculator)
            : base(peptideMods, peptideSeqMassCalculator)
        {
        }

        public override void Clear()
        {
            base.Clear();

            MostAbundantIsotopeMz = string.Empty;
            Modifications = string.Empty;
            Composition = string.Empty;
            ProteinDesc = string.Empty;
            ProteinLength = string.Empty;
            ResidueStart = string.Empty;
            ResidueEnd = string.Empty;
            MatchedFragments = string.Empty;
            SpecEValue = string.Empty;
            EValue = string.Empty;
            QValue = string.Empty;
            PepQValue = string.Empty;
        }
    }
}
